sphinx/domains/python.py

```diff
@@ -517,7 +521,7 @@ def run(self):
 class PyXRefRole(XRefRole):
     def process_link(self, env, refnode, has_explicit_title, title, target):
         refnode['py:module'] = env.ref_context.get('py:module')
-        refnode['py:class'] = env.ref_context.get('py:class')
+        refnode['py:class_stack'] = env.ref_context.get('py:class_stack')
```

Comment:
Here is where I said "the key of nodes"
I think there are no reason to change tye key of reference node (`refnode`).
In this case, only key of `ref_context` should be changed.